### DISCUSSION TITLE
Mini Effects Stack on Track Headers

### DISCUSSION THREAD

**Discussion by hyuri** [audacity/audacity#4223] [Ideas & Feature Requests]
### Your idea

Mini effects stack built into the track header itself, for quick opening/adding/deleting/disabling/enabling of effects.

Behaviour:
1. Left Clicking on an empty rack on the mini stack opens a dialog to choose/search effect to add;
2. Left Clicking on an effect on the track opens a window with that effect;
3. Shift + Left Clicking on an effect disables the effect, dims and colors the button for visual feedback;
4. Ctrl/Cmd + Shift + Left Clicking on an effect makes it offline and dims the button for visual feedback;
5. Ctrl/Cmd + Left Clicking opens a window with that effect but also the full stack;
6. Alt/Option + Left Clicking an effect deletes the effect;
7. Drag & Drop effect up and down the mini stack to reorder the stack;
8. Scrolling up and down the mini stack scrolls it up and down when there are more effects on the stack than can be displayed;
9. Right Clicking on an effect or empty rack on the mini stack brings up a context menu with options such as "Add effect", "Replace effect", "Delete effect", "Disable/Enable effect", etc.

In companion to this mini stack, an "Effects" button that you can click to bring up the full stack. Shift + Left Clicking on this "Effects" button disables/enables (toggles) the entire effects stack.

![Audacity - Mini Effects Stack on Track Header](https://user-images.githubusercontent.com/18584014/213683645-7973d54d-f3df-4855-b439-472001de8c8f.jpg)

The green headphones button here is meant to be input monitoring for that track.

I think the "Effects" button shown here could even be split into two buttons — "FX" and "routing icon" — and have the routing icon bring up routing options, but that's outside of the scope of this particular feature request.

### Problem to be solved

We get an at-a-glance overview of the effects attached to each track, and we also reduce the number of clicks and mouse travel distance necessary to perform basic effect stack operations.

### Prior art

Pro Tools has this. Reaper has it as well. Both I think are optional but, if I recall correctly, it's the default in Pro Tools.

**Comment by LWinterberg** [comment]
This proposal runs into the problem of having the track header become very cluttered very quickly, which is a problem which DAWs tend to have in general. Fwiw, we're currently looking into both having the effects panel display the effects of multiple tracks, as well as a mixer with this kind of capability

**Reply by hyuri** [reply]
True. But are there any downsides? Maybe a bit overwhelming for beginner users? It's very useful in other DAWs, it saves a step multiplied by the number of times you open effects to tweak.

**Comment by petersampsonaudacity** [comment]
See also this previously logged Enhancement Request:   
**There is no visual (or audible) cue in the TCP for an invisible active Effects Stack #4092**
